### PR TITLE
MM-15116 Migrate WebHook.SaveIncoming Method to Sync by default

### LINKED ISSUE #10616: [Help Wanted] [MM-15116] Migrate "WebHook.SaveIncomming" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15116)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `SaveIncomming` in the `WebHook` store.

The expected way to implement it is, go to the `WebHook` store implementation in the `store/sqlstore/` directory, modify the method `SaveIncomming` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by andresoro**
#### Summary

Migrates the WebHook.SaveIncoming methods to sync by default. I was previously working on this issue in PR #10635 but I made this mistake of adding to many changes. This PR Is only for the SaveIncoming method.

#### Ticket Link
Fixes #10616

### CODE REVIEW

File: app/webhook.go [hunk 1]
```diff
@@ -323,13 +323,8 @@ func (a *App) CreateIncomingWebhookForChannel(creatorId string, channel *model.C
...
 	}
 
-	if result := <-a.Srv.Store.Webhook().SaveIncoming(hook); result.Err != nil {
-		return nil, result.Err
-	} else {
-		return result.Data.(*model.IncomingWebhook), nil
-	}
+	return a.Srv.Store.Webhook().SaveIncoming(hook)
 }
-
```
Comment (jespino) [inline app/webhook.go]: nit: expected 1 blank line between functions

**Comment by jespino** [comment]
@andresoro feel free to submit the other 2 whenever you want, we don't mind to have many PRs open at the same time, the problem is with PR to big to be reviewed.

**Comment by andresoro** [comment]
@jespino hey, any ideas on why the build is failing now? It was passing CI when I first submitted.

**Comment by andresoro** [comment]
Hey @jespino, any ideas on why the build is failing now after it initially passed the CI pipeline?

**Comment by andresoro** [comment]
I figured it out. A test case in storetest/webhook_store.go was changed on master that used async version of WebHook.SaveIncoming method. I updated the test case accordingly.

**Pull request merged**